### PR TITLE
ACQ-342 Implement intro offer message

### PR DESCRIPTION
**Pull request by nickramsbottom**
Ticket: https://financialtimes.atlassian.net/browse/ACQ-342
Design: https://marvelapp.com/prototype/24j3g43g/screen/72774236
Depends on & screenshots: https://github.com/Financial-Times/next-messaging-guru/pull/109

### CODE REVIEW

File: client/components/top/try-full-access/main.scss [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+			}
+
+			.o-message__content-main {
+				display: flex;
+				flex-direction: column;
+
+				margin-top: oSpacingByName('s4');
+				margin-bottom: oSpacingByName('s6');
+				text-transform: uppercase;
+				font-size: 18px;
```
Comment (nickramsbottom) [inline client/components/top/try-full-access/main.scss:45]: we might be able to use o-fonts for this: https://registry.origami.ft.com/components/o-fonts@4.4.1/

File: client/components/top/try-full-access/main.scss [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+					margin-bottom: oSpacingByName('s8');
+					font-size: 24px;
+					line-height: 28px;
+				}
+			}
+		}
+
+		.o-message__content-detail {
+			color: oColorsByName('white');
+			.highlight {
```
Comment (nickramsbottom) [inline client/components/top/try-full-access/main.scss:60]: is this rule covered by `o-message__content-highlight`?

File: manifest.js [hunk 3]
```diff
@@ -79,5 +79,10 @@ module.exports = {
 	remainingArticlesTest: {
 		path: 'bottom/remaining-articles',
 		lazy: true
-	}
+	},
+	tryFullAccess: {
+		path: 'top/lazy',
+		lazy: true,
+		guruQueryString: 'offerId=683995bf-ed3f-2fd6-d589-1628ebad1629',
```
Comment (nickramsbottom) [inline manifest.js:86]: this needs to be the production offer ID:
```suggestion
		guruQueryString: 'offerId=41218b9e-c8ae-c934-43ad-71b13fcb4465',
```

File: client/components/top/try-full-access/main.scss [hunk 4]
```diff
@@ -0,0 +1,102 @@
+@include oMessageAddState('try-full-access', (
+	'foreground-color': 'white',
```
Comment (mifkata) [inline client/components/top/try-full-access/main.scss:2]: I think you need to use `oColorsByName` to be consistent with the rest of the codebase

**Review by nickramsbottom — COMMENTED** [review]
Nice! Small changes

File: client/components/top/try-full-access/main.scss (hunk 4 above)
Comment (nickramsbottom) [reply client/components/top/try-full-access/main.scss:2]: good catch! examples here: https://github.com/Financial-Times/o-banner/blob/9579cf3211f7ac4b5506214e9a56ade949cbf7ee/demos/src/demo.scss#L28

Comment (valeksiev) [reply client/components/top/try-full-access/main.scss:2]: The value provided is passed through `oColorsByName` by the [mixin here](https://github.com/Financial-Times/o-message/blob/f2caafb8475989b1f7664f0c300594809bec5211/src/scss/_state.scss#L19). It works both ways of course but is redundant. Do we want to be on the safe side?

Also in `n-messaging-client/client/components/o-mssage-notice.scss` (which I used as an example) string values are used. Do we unify when we see inconsistencies like this?

File: client/components/top/try-full-access/main.scss (hunk 1 above)
Comment (valeksiev) [reply client/components/top/try-full-access/main.scss:45]: I checked this page while working on this didn't find anything about the font size. Seems like `oFonts` is used only to load the font but the actual use is plain ol CSS.

Also in all other `scss` files in the project, the font-sizes are specified like this.

File: client/components/top/try-full-access/main.scss (hunk 2 above)
Comment (valeksiev) [reply client/components/top/try-full-access/main.scss:60]: I am not sure I understand the questions but this is what makes the first word of the second row pink. 😆

File: client/components/top/try-full-access/main.scss [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+			@include oGridRespondTo(S) {
+				margin-left: 70px;
+			}
+			@include oGridRespondTo(M) {
+				margin-left: 132px;
+			}
+		}
+
+		.o-message__content-highlight {
+			color: oColorsByName('wheat');
```
Comment (nickramsbottom) [inline client/components/top/try-full-access/main.scss:77]: can this be specified in `oMessageAddState`? line 4

File: client/components/top/try-full-access/main.scss (hunk 2 above)
Comment (nickramsbottom) [reply client/components/top/try-full-access/main.scss]: my comment on the other PR might make it clearer: https://github.com/Financial-Times/next-messaging-guru/pull/109#discussion_r490848138

File: client/components/top/try-full-access/main.scss (hunk 4 above)
Comment (nickramsbottom) [reply client/components/top/try-full-access/main.scss]: it's worth unifying, yes! good spot

File: client/components/top/try-full-access/main.scss (hunk 5 above)
Comment (valeksiev) [reply client/components/top/try-full-access/main.scss:77]: `oMessageAddState` updates both the highlight color and the button color so we need to override one of those. From the docs:
```
	'highlight-color': 'lemon', // lemon highlights with `o-message__content-highlight-color` and a lemon button
```

Comment (nickramsbottom) [reply client/components/top/try-full-access/main.scss:72]: ahhh nice!

**Review by nickramsbottom — COMMENTED** [review]
I can't approve my own PR but changes on top of mine look good!

**Review by mifkata — APPROVED** [review]
LGTM

**Pull request merged**